Update classic mania score import formula

Adrriii opened the pull request:
This preserves the integrity of mania rankings

Since it's provably impossible to statically convert scorev1 to scorev2 in mania, the best we can do is a rough estimation, however a single multiplier is a bit insensible. I'm aware this PR doesn't impact stable score import for leaderboards ; But I can only guess that it is done in the same way in your routines. So this is an attempt at having them match this logic.

As a side effect, this makes the "Classic" mode have no multiplier, due to it being used in the stable import tool. Maybe this isn't necessary? The most important matter being actual online leaderboards.

`Instead of the default 0.96 classic mod multiplier, use a slightly better formula to reflect high scores' transition from v1 to v2. Keep a maximum nerf of around 95% of original score to reflect most scorev2 scores compared to scorev1.`

This is a voluntarily simple change, as it might not be necessary to go in further complications for this matter.

![image](https://github.com/ppy/osu/assets/16978299/119a9e89-0bd3-43a8-907d-f8f269ddbbb5)

bdach commented:
I don't understand what this is doing or why. What is wrong precisely with the current method?

Adrriii commented:
> What is wrong precisely with the current method?

The current conversion for mania is resulting from the std mod multiplier, and a deduced accuracy portion. Meaning every stable score will be capped at 960k. In mania, it is incredibly easy to surpass such a cap, so much that any map (except for the very hardest maps), will see its scores immediately surpassed without any effort. This is roughly equivalent to not importing scores from a leaderboard standpoint.

I can understand if this is not a concern, however such a minor fix could have a great effect on transitioning leaderboards. Scores that should be beatable will have a decent margin to be. (a 50k margin below 950k, and a two-fold gap towards 1M for higher scores.

> What are the graphs showing even?

The graphs are admittedly badly annotated : The blue line (scorev1) represents a stable score for any given x and the red line is its equivalent with the proposed implementation. Should help visualizing how leaderboards could be, at least temporarily, well preserved.

> It does actually. The code you modified is responsible for converting stable scores to new standardised. Therefore it needs to match the new standardised scoring implementation. Which it did before your change.

Doesn't it only matter if the output score is valid ? Not all parameters should be used if we're only talking about stable conversion, since the only one that's actually holding value in regards to scorev1 scoring, is the combo portion.

Adrriii committed:
Update classic mania score import formula

Instead of the default 0.96 classic mod multiplier, use a slightly better formula to reflect high scores' transition from v1 to v2. Keep a maximum nerf of around 95% of original score to reflect most scorev2 scores compared to scorev1.

bdach commented:
> Meaning every stable score will be capped at 960k. In mania, it is incredibly easy to surpass such a cap, so much that any map (except for the very hardest maps), will see its scores immediately surpassed without any effort. This is roughly equivalent to not importing scores from a leaderboard standpoint.

If this is your concern, then your change to classic multiplier suffices to address it.

> Doesn't it only matter if the output score is valid ? Not all parameters should be used if we're only talking about stable conversion, since the only one that's actually holding value in regards to scorev1 scoring, is the combo portion.

The conversion should approximate *current lazer scoring* as close as possible. In order for imported stable scores to be on the same footing as lazer scores. Deviating from that effectively leads to two scoring systems being in effect: one for stable scores and one for mania scores. This renders the conversion process pointless. If we wanted to do that, we just wouldn't recompute any scores.

Adrriii commented:
> your change to classic multiplier suffices to address it.

If that is so, then it might be best to only keep this change then.

> The change to the conversion formula is in my eye 100% unfounded and wrong.

Reverting this

Adrriii committed:
Keep accuracy based conversion without Classic multiplier

Adrriii commented:
Done, if i understand correctly then, with the accuracy formula part, the nerf should already be enough to keep leaderboards going in v2

peppy commented:
I think at this point, it'd best you open a discussion (or add to one of the many existing ones) explaining your thoughts.

Adrriii commented:
> it'd best you open a discussion

I've then just done so, going in even more lengths

> or add to one of the many existing ones

Couldn't find a relevant discussion to append these to, except maybe the Survey since there were results about the 1M cap but this isn't really related to stable import. And would crowd it too much imho. The rest of the discussions are just skinning related.
